centreon/src/Core/Media/Infrastructure/Repository/DbReadMediaRepository.php

```diff
@@ -244,20 +239,159 @@ private function loadDatabase(): void
     }
 
     /**
-     * @return callable
+     * @inheritDoc
      */
-    private function createMedia(): callable
+    public function findByRequestParameters(RequestParametersInterface $requestParameters): \Countable&\Iterator
     {
-        return function (array $data) {
-            $media = new Media(
-                $data['img_id'],
-                $data['img_path'],
-                $data['dir_name'],
-                null
-            );
-            $media->setComment($data['img_comment']);
-
-            return $media;
+        return new class ($this->db, $this->createMedia(...), $requestParameters) extends AbstractRepositoryRDB
```

Comment:
I think we should have used [IteratorAggregate](https://www.php.net/manual/en/class.iteratoraggregate.php) by default which is far easier to test & maintain.

But anyway, if it works ... :shrug: